app/helpers/local_time_helper.rb

```diff
@@ -12,9 +12,15 @@ def local_time(time, options = nil)
   end
 
   def local_date(time, options = nil)
+    time = utc_time(time)
+
     options, format = extract_options_and_value(options, :format)
-    options[:format] = format || LocalTime.default_date_format
-    local_time time, options
+    format = find_date_format(format)
+
+    options[:data] ||= {}
+    options[:data].merge! local: :time, type: :date, format: format
+
+    time_tag time, time.strftime(format), options
```

Comment:
I think we can still call `local_time` here if we calculate the correct format in `#local_date` and pass it as a string (as opposed to a symbol) to `local_time`.